src/components/account/MarketWithdrawal.tsx

```diff
@@ -63,24 +63,44 @@ export function MarketWithdrawal({
     withdrawAmount === 0n ||
     withdrawAmount > marketBalance;
 
+  const isMaxDisabled = withdrawStatus.state === TransactionState.Loading || marketBalance <= 0n;
+
   const isZero = withdrawAmount <= 0n && isFocused;
 
   const isTooLarge = withdrawAmount > 0n && BigInt(withdrawAmount) > marketBalance && isFocused;
 
+  const setMaxAmount = () => {
+    setWithdrawAmount(marketBalance);
+  };
+
   return (
     <div className="flex-1 min-w-0 flex flex-col gap-2">
       <h3 className="text-lg font-semibold">💰 Withdraw Market Balance</h3>
       <p className="text-sm text-gray-600">Enter the amount to withdraw (Planck units).</p>
       <div className="flex items-center gap-2">
-        <input
-          type="number"
-          value={withdrawAmount.toString()}
-          onChange={(e) => setWithdrawAmount(BigInt(e.target.value))}
-          onFocus={() => setIsFocused(true)}
-          onBlur={() => setIsFocused(false)}
-          placeholder="Amount in Planck"
-          className="px-3 py-2 border rounded text-sm"
-        />
+        <div className="relative">
+          <input
+            type="number"
+            value={withdrawAmount.toString()}
+            onChange={(e) => setWithdrawAmount(BigInt(e.target.value))}
+            onFocus={() => setIsFocused(true)}
+            onBlur={() => setIsFocused(false)}
+            placeholder="Amount in Planck"
+            className="w-full px-3 py-2 border rounded text-sm pr-16"
+          />
+          <button
+            type="button"
+            disabled={isMaxDisabled}
+            onClick={setMaxAmount}
+            className={`absolute right-1 top-1/2 transform -translate-y-1/2 px-2 py-1 rounded text-xs transition ${
+              isMaxDisabled
+                ? "bg-gray-300 cursor-not-allowed"
+                : "bg-blue-600 text-white hover:bg-blue-700"
+            }`}
+          >
+            MAX
+          </button>
```

Comment:
```suggestion
{!isMaxDisabled && (
            <button
              type="button"
              onClick={setMaxAmount}
              className={
                "absolute right-1 top-1/2 transform -translate-y-1/2 px-2 py-1 rounded text-xs transition bg-blue-600 text-white hover:bg-blue-700"
              }
            >
              MAX
            </button>
          )}
```
I would condtionally render the button based on the withdrawState and the market balance instead of graying it out.

Reply:
I agree that disabling the button would confuse the user. But I also think that it's confusing if the button is shown sometimes and not shown at other times. So I've changed the button to never hide or disable. So in cases when the user has zero available on the market. If they click on the MAX. The zero is used as an amount. Which I think is fine because 0 is practically the max amount that can be withdrawn. What do you think?

Backend devs debating what is better from the UX perspective :D

Reply:
> Backend devs debating what is better from the UX perspective :D

😂 😂 


Having it always be there is a good solution too, since 0 can also be the max when there is no balance.